File: app/src/main/java/com/sp/base/util/LocationUtil.java
package com.sp.base.util;

import android.content.Context;
import android.location.Location;

import java.text.DecimalFormat;

public class LocationUtil {
    // For getting distance between two locations
    public static String getDistance(Context context, String sLatitude, String sLongitude, String dLatitude, String dLongitude) {

        if (!dLatitude.equals("") && !dLongitude.equals("")) {
            double mLatitude = 0;
            double mLongitude = 0;
            try {
                mLatitude = Double.parseDouble(dLatitude);
                mLongitude = Double.parseDouble(dLongitude);
            } catch (NumberFormatException e) {
                e.printStackTrace();
            }

            Location locationA = new Location("point A");
            if (!sLongitude.equals("")) {
                try {
                    locationA.setLatitude(Double.parseDouble(sLatitude));
                    locationA.setLongitude(Double.parseDouble(sLongitude));
                } catch (NumberFormatException e) {
                    e.printStackTrace();
                }
            }

            Location locationB = new Location("point B");
            locationB.setLatitude(mLatitude);
            locationB.setLongitude(mLongitude);

            double distance = locationA.distanceTo(locationB) / 1609;

            String mDistanceStr = new DecimalFormat("###.##").format(distance) + " mi";
            return mDistanceStr.replace(",", ".");
        }
        return "";
    }

}

File: app/src/main/java/com/baseapplication/util/LocationUtil.kt
package com.baseapplication.util

import android.content.Context
import android.location.Location
import java.text.DecimalFormat

object LocationUtil {
    // For getting distance between two locations
    fun getDistance(
        context: Context?,
        sLatitude: String,
        sLongitude: String,
        dLatitude: String,
        dLongitude: String
    ): String {
        if (!(dLatitude == "") && !(dLongitude == "")) {
            var mLatitude: Double = 0.0
            var mLongitude: Double = 0.0
            try {
                mLatitude = dLatitude.toDouble()
                mLongitude = dLongitude.toDouble()
            } catch (e: NumberFormatException) {
                e.printStackTrace()
            }
            val locationA: Location = Location("point A")
            if (!(sLongitude == "")) {
                try {
                    locationA.setLatitude(sLatitude.toDouble())
                    locationA.setLongitude(sLongitude.toDouble())
                } catch (e: NumberFormatException) {
                    e.printStackTrace()
                }
            }
            val locationB: Location = Location("point B")
            locationB.setLatitude(mLatitude)
            locationB.setLongitude(mLongitude)
            val distance: Double = (locationA.distanceTo(locationB) / 1609).toDouble()
            val mDistanceStr: String = DecimalFormat("###.##").format(distance) + " mi"
            return mDistanceStr.replace(",", ".")
        }
        return ""
    }
}
